include/nbl/builtin/hlsl/shapes/beziers.hlsl

```diff
@@ -40,89 +40,204 @@ namespace shapes
-        float tForMajorCoordinate(const int major, float x) 
+        float_t tForMajorCoordinate(const int major, float_t x) 
         { 
-            float a = A[major] - x;
-            float b = B[major] - x;
-            float c = C[major] - x;
+            float_t a = P0[major] - x;
+            float_t b = P1[major] - x;
+            float_t c = P2[major] - x;
             int rootCount;
-            float2 roots = SolveQuadratic(float3(a, b, c), rootCount);
+            float2_t roots = SolveQuadratic(float3_t(a, b, c), rootCount);
             // assert(rootCount == 1);
             return roots.x;
         }
+        
+        float_t calcArcLen(float_t t, ArcLengthPrecomputedValues preCompValues)
+        {
+            float_t lenTan = sqrt(t*(preCompValues.a*t+ preCompValues.b)+ preCompValues.c);
+            float_t retval = 0.5f*t*lenTan;
+            float_t sqrtc = sqrt(preCompValues.c);
+            
+            // we skip this because when |a| -> we have += 0/0 * 0 here resulting in NaN
+            if (preCompValues.lenA2>=exp2(-23.f))
+                retval += preCompValues.b_over_4a*(lenTan-sqrtc);
+
+            // sin2 multiplied by length of A and B
+            float det_over_16 = preCompValues.AdotB* preCompValues.AdotB- preCompValues.lenA2* preCompValues.c;
+            // because `b` is linearly dependent on `a` this will also ensure `b_over_4a` is not NaN, ergo `a` has a minimum value
+            if (det_over_16>=exp2(-23.f))
+            {
+            // TODO: finish by @Przemog
+                //retval += det_over_16*...;
+            }
+
+            return retval;
+        }
+
+        float_t calcArcLenInverse(float_t arcLen, float_t accuracyThreshold, float_t hint, ArcLengthPrecomputedValues preCompValues)
+        {
+            float_t xn = hint;
+
+            if (arcLen <= accuracyThreshold)
+                return arcLen;
+
+            float2_t B = 2.0*(P1 - P0);
+            float2_t twoA = 2.0*(P2 - P1) - B;
+
+            const int iterationThreshold = 32;
+            for(int n = 0; n < iterationThreshold; n++)
+            {
+                float_t arcLenDiffAtParamGuess = arcLen - calcArcLen(xn, preCompValues);
+
+                if (abs(arcLenDiffAtParamGuess) < accuracyThreshold)
+                    return xn;
+
+                float_t differentialAtGuess = length(twoA * xn + B);
+                    // x_n+1 = x_n - f(x_n)/f'(x_n)
+                xn -= (calcArcLen(xn, preCompValues) - arcLen) / differentialAtGuess;
+            }
+
+            return xn;
+        }
     };
 
-    struct QuadraticBezierOutline
+    template<typename float_t>
+    struct Quadratic
     {
-        QuadraticBezier bezier;
-        float thickness;
-
-        static QuadraticBezierOutline construct(float2 a, float2 b, float2 c, float thickness)
+        using float2_t = vector<float_t, 2>;
+        using float3_t = vector<float_t, 3>;
+        
+        struct ArcLengthPrecomputedValues
```

Comment:
make a comment here
```
// These are precompured values to calculate arc length which eventually is an integral of sqrt(a*x^2+b*x+c)
```
@devshgraphicsprogramming Do I understand this correctly that these a,b,c's here for the precomputed values are for integrating ` sqrt(a*x^2+b*x+c)`?

Reply:
done